org.eclipse.jdt.debug.ui/ui/org/eclipse/jdt/internal/debug/ui/actions/ToggleBreakpointAdapter.java

```diff
@@ -1894,4 +1952,110 @@ private static ICompilationUnit getCompilationUnit(ITextEditor editor) {
 		return JavaCore.createCompilationUnitFrom(file);
 	}
 
+	public int selectLambda(List<LambdaExpression> lambdaExps) {
+		if (lambdaExps.isEmpty()) {
+			return -1;
+		}
+
+		int[] selection = { -1 };
+		Shell parentShell = PlatformUI.getWorkbench().getActiveWorkbenchWindow().getShell();
+		Shell popup = new Shell(parentShell, SWT.TOOL | SWT.RESIZE);
+		popup.setText(ActionMessages.lambdaSelection);
+		GridLayout popupLayout = new GridLayout(1, false);
+		popupLayout.marginWidth = 10;
+		popupLayout.marginHeight = 10;
+		popupLayout.verticalSpacing = 10;
+		popup.setLayout(popupLayout);
+
+		Table table = new Table(popup, SWT.BORDER | SWT.V_SCROLL | SWT.H_SCROLL);
+		GridData tableData = new GridData(SWT.FILL, SWT.FILL, true, true);
+		table.setLayoutData(tableData);
+		List<String> itemsName = toLambdaEntries(lambdaExps);
+
+		for (String item : itemsName) {
+			new TableItem(table, SWT.NONE).setText(item);
+		}
+
+		table.addMouseListener(new MouseAdapter() {
+			@Override
+			public void mouseDoubleClick(MouseEvent e) {
+				int ind = table.getSelectionIndex();
+				if (ind >= 0) {
+					selection[0] = ind;
+					popup.close();
+				}
+			}
+		});
+
+		Composite buttonBar = new Composite(popup, SWT.NONE);
+		GridLayout buttonLayout = new GridLayout(2, true);
+		buttonLayout.marginWidth = 0;
+		buttonLayout.marginHeight = 0;
+		buttonLayout.horizontalSpacing = 10;
+		buttonBar.setLayout(buttonLayout);
+		buttonBar.setLayoutData(new GridData(SWT.END, SWT.CENTER, true, false));
+		Button selectButton = new Button(buttonBar, SWT.PUSH);
+		selectButton.setText(ActionMessages.lambdaSelect);
+		selectButton.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, true, false));
+
+		selectButton.addListener(SWT.Selection, e -> {
+			int index = table.getSelectionIndex();
+			if (index >= 0) {
+				selection[0] = index;
+				popup.close();
+			}
+		});
+		Button closeButton = new Button(buttonBar, SWT.PUSH);
+		closeButton.setText(ActionMessages.lambdaClose);
+		closeButton.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, true, false));
+		closeButton.addListener(SWT.Selection, e -> popup.close());
+		int tableHeight = Math.min(lambdaExps.size(), 10) * table.getItemHeight() + 80;
+		popup.setSize(300, tableHeight);
```

Comment:
still lot of hard coded values for widget sizes etc. Whole manual creation should be replaced with existing `ElementListSelectionDialog` dialog.